### ISSUE TITLE
axis for unit quaterion given as (0, 0, 0)

### ISSUE THREAD

**Issue by jodyxha** [KieranWynn/pyquaternion#86]
Hi
I noticed that the axis of the unit quaternion (1,0,0,0) is returned as (0,0,0).

This code

    from pyquaternion import Quaternion
    
    q=Quaternion()`
    print("q: %s"%q)`
    print("axis %s, angle %f"%(q.axis,q.angle))`

produces this output:

    q: 1.000 +0.000i +0.000j +0.000k`
    axis [0. 0. 0.], angle 0.000000`

I think this behavior is not correct. 
Of course, for an angle of 0 the axis is arbitrary, but it should not be [0,0,0]

**Comment by teodron** [comment]
The `[0,0,0]` is indeed not a unit vector. The axis is thus undetermined. Perhaps it would be best to indicate this situation through `nan` values instead of zeros and leave it up to the user of the library to indicate how they wish to proceed further. Another option would be through a special policy, e.g. offering a class property of the Quaernion class that users can override to their wishes, a property that could be named `UndeterminedAxis` or similar. By default, that could return `nan`s instead of zeros, but the whole idea would be to have users configure the behavior. I also treated such an edge case a while ago by simply choosing `1,0,0` as an axis with no other consequence on my algorithm, but that is not a mathematically generally correct or without consequences. Thus, users should know that by default they will encounters nan values, but they can choose to tune the behavior if they really need a specific workaround and do not want to handle nans explicitly.

**Comment by jodyxha** [comment]
Yes, I have done a similar workaround, but it did not seem very clean :).
I also think the returning of nan values would be a good option.
